Library/ViewModels/AddNewCardViewModel.swift

```diff
@@ -39,16 +41,53 @@ AddNewCardViewModelOutputs {
 
   public init() {
     let cardholderName = self.cardholderNameChangedProperty.signal.skipNil()
-    let creditCardDetails = self.creditCardChangedProperty.signal.skipNil()
+    let creditCardDetails = self.creditCardChangedProperty
+      .signal
+      .skipNil()
+      .filterMap { args -> (String, UInt, UInt, String)? in
+        let (cardNumberArg, expMonthArg, expYearArg, cvcArg) = args
+
+        guard let expMonth = expMonthArg,
+              let expYear = expYearArg,
+              let cvc = cvcArg else {
+          return nil
+        }
+
+        return (cardNumberArg, expMonth, expYear, cvc)
+      }
+
+    let cardNumber = self.creditCardChangedProperty
+      .signal
+      .skipNil()
+      .map { $0.0 }
 
     self.cardholderNameBecomeFirstResponder = self.viewDidLoadProperty.signal
     self.paymentDetailsBecomeFirstResponder = self.cardholderNameTextFieldReturnProperty.signal
 
+    let cardBrandValidAndCardNumberValid = Signal
+      .combineLatest(self.cardBrandIsValidProperty.signal,
+                     cardNumber)
+      .map { (brandValid, cardNumber) -> Bool in
+        // If card number is insufficiently long, always return "valid card brand" behaviour
+        if cardNumber.count < 2 {
```

Comment:
I like the second option since it's more optimized to return `true` if the first statement is `true` (I think it won't even need to evaluate the second statement)